documentation/design/proposals/bridging_external.adoc

```diff
@@ -0,0 +1,80 @@
+    remoteAddress: foo/queue1 
+    direction: in | out # Specifies if messages are forwarded from localAddress to remoteAddress or from remoteAddress to localAddress.
+```
+
+NOTE: The localAddress of a forwarder may not exist locally at the point of creating the address space. A forwarding rule will only be applied if the localAddress exists and if the remoteAddress is matched by any of the entries in the connector `addresses` field. Sanity checking on the forwarders and connectors is needed to ensure that invalid forwarding rules cannot be applied.
+
+The sending and receiving messages to a remote AMQP endpoint via local address space use cases are covered by simply creating a connector and attaching client to 'connectorname/address'.
+
+The forwarding use cases are covered by creating a connector and a forwarder referencing the connector. The direction field will determine if messages are forwarded from remote to local or from local to remote.
+
+
+== Implementation
+
+==== Api server
+
+The api server will need to perform an extra set of validation when creating address spaces to ensure that references from forwarder to connector are correct and that remoteAddress matches an address in the connector. The api-server should not require remoteAddress in forwarders to exist because it may not have been created yet.
+
+==== Address space controller
+
+The address space controller will need to pass additional configuration to the router when starting up. When creating the address space, the router config configmap need to be programatically crafted in order for the additional information to be optionally added to the config. The connector and forwarder definitions will essentially end up defining a set of connectors and autoLinks in the router config.
+
+This kind of fine-grained reconsiliation is a direction we're likely to go in when moving the address-space-controller logic to the controller-manager (enmasse-operator) in the future, but will create additional code that have to be rewritten if we implement this in the address-space-controller.
+
+To avoid that, as an alternative, we could put the responsibility of creating the router config configmap to the controller-manager as a first step in this direction (the configmap would then have to be removed from the template). From a maintenance POV it it might be better to have the logic in the same 'place' though, and the controller-manager would not be aware of any processing error in the address-space-controller such as failing to create the address space, in which case one would end up with a stale configmap.
+
+== Testing
+
+* Set up an AMQP broker (i.e. Artemis) on a remote host
+* Configure an address space with connectors and forwarders for the 4 different use cases listed initially
+
+== Documentation
+
+The addifional fields should be added to the address space reference, and the tenant guide should be extended to cover the use cases as a realistic scenario with examples.
```

Comment:
I think the existing authorisation scheme we have for address (as implemented in standard) should work with your proposal without change.  The user could express patterns for remote queues `remote1/foo*` or `remote1/foobar`

Reply:
Yes, that is the intention. I'll add a note about it.